lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -227,7 +227,7 @@ def __get_k8s_job_spec_template(self, ajs):
         (see pod selector) and an appropriate restart policy."""
         k8s_spec_template = {
             "metadata": {
-                "labels": {"app": self.__produce_unique_k8s_job_name(ajs.job_wrapper.get_id_tag())}
+                "labels": {"app": self.__produce_unique_k8s_job_name(ajs.job_wrapper.get_id_tag())[:-5]}
```

Comment:
Haven't elevated this from draft pending a cleaner solution here - stripping the last 5 characters for the app label works but is hard to understand.